### DISCUSSION TITLE
A better version of helpsleeper

### DISCUSSION THREAD

**Discussion by Ng76i4** [mpurses/Sonder#9] [Show and tell]
`
[Rainmeter]
Update=500
AccurateText=1

[Metadata]
Name=Sleep Helper
Author=NG76i4
Information=Automatic load\unload Visualizer which help the computer sleep while idle
License=Creative Commons Attribution-Non-Commercial
Version=1.0.3

; I know not thing about coding. I just studied from rainmeter manual, copied here and there. This thing may have some mistake but at least it works.

[Variables]
@include=#@#Variables.inc

[MeasureState]
Measure=Plugin
Plugin=WebNowPlaying
PlayerType=State
IfCondition=MeasureState = 1
IfTrueAction=[!ActivateConfig "#ROOTCONFIG#\Visualizer" "Dots-Visualizer.ini"][!Log "Spotify Playing: Load Visualizer"]
IfCondition2=(MeasureState = 0)
IfTrueAction2=[!DeactivateConfig "#ROOTCONFIG#\Visualizer"][!Log "Spotify Closed: Unload Visualizer"]

[RequiredMeter]
Meter=String
Disabled=1
Text=""
`

**Comment by mpurses** [comment]
I assume you aren't using Spotify?  The WebNowPlaying plugin will definitely work best if you are using a browser for your music.  I will eventually support the WebNowPlaying plugin, but I didn't include it yet as I was debating making my own plugin/chrome extension to better catch all audio playing sources so the Visualizer is only enabled when it should be.

If not (you are using Spotify), what issues were you having with the my version of the SleepHelper?

**Reply by Ng76i4** [reply]
I am only use the Spotify desktop application on windows

No. There are no issues at all. I found that your sleephelper only unload the visualizer when I turn off the spotify in windows tray. And because I rarely turn the Spotify completely off so I think it will be better if the visualizer unload when I pause or stop the player.
By adding "If Condition2=(MeasureState = 0) || ((MeasureState = 2)".

Sorry for lack of knowledge of coding and stuffs. I just learn and copy from the rainmeter manual
https://docs.rainmeter.net/manual/measures/general-options/ifconditions/

**Reply by mpurses** [reply]
Gotcha, that makes sense and your workflow helps me out. I'll include that condition/webnowplaying plugin in the next version.
